common/src/main/java/io/netty/util/internal/PlatformDependent.java

```diff
@@ -921,92 +910,19 @@ private static boolean isRoot0() {
-                    try {
-                        int exitCode = p.waitFor();
-                        if (exitCode != 0) {
-                            uid = null;
-                        }
-                        break;
-                    } catch (InterruptedException e) {
-                        // Ignore
-                    }
-                }
-            } catch (Throwable ignored) {
-                // Failed to run the command.
-                uid = null;
-            } finally {
-                if (in != null) {
-                    try {
-                        in.close();
-                    } catch (IOException e) {
-                        // Ignore
-                    }
-                }
-                if (p != null) {
-                    try {
-                        p.destroy();
-                    } catch (Exception e) {
-                        // Android sometimes triggers an ErrnoException.
-                    }
-                }
-            }
-
-            if (uid != null && UID_PATTERN.matcher(uid).matches()) {
-                logger.debug("UID: {}", uid);
-                return "0".equals(uid);
-            }
-        }
-
-        logger.debug("Could not determine the current UID using /usr/bin/id; attempting to bind at privileged ports.");
-
-        Pattern PERMISSION_DENIED = Pattern.compile(".*(?:denied|not.*permitted).*");
-        for (int i = 1023; i > 0; i --) {
-            ServerSocket ss = null;
-            try {
-                ss = new ServerSocket();
-                ss.setReuseAddress(true);
-                ss.bind(new InetSocketAddress(i));
-                if (logger.isDebugEnabled()) {
-                    logger.debug("UID: 0 (succeded to bind at port {})", i);
-                }
-                return true;
-            } catch (Exception e) {
-                // Failed to bind.
-                // Check the error message so that we don't always need to bind 1023 times.
-                String message = e.getMessage();
-                if (message == null) {
-                    message = "";
-                }
-                message = message.toLowerCase();
-                if (PERMISSION_DENIED.matcher(message).matches()) {
-                    break;
-                }
-            } finally {
-                if (ss != null) {
-                    try {
-                        ss.close();
-                    } catch (Exception e) {
-                        // Ignore.
-                    }
+        try {
+            final String username = AccessController.doPrivileged(new PrivilegedAction<String>() {
+                @Override
+                public String run() {
+                    return System.getProperty("user.name");
                 }
-            }
+            });
+            // Check for root and toor as some BSDs have a toor user that is basically the same as root.
+            return "root".equals(username) || "toor".equals(username);
```

Comment:
seems like this could be fooled into thinking you have root privileges root when you do not. what if someone has a user name `toor`? Is there a better alternative to this?